src/core/gosling.schema.ts

```diff
@@ -358,100 +511,265 @@ export type Aggregate = 'max' | 'min' | 'mean' | 'bin' | 'count';
 /* ----------------------------- DATA ----------------------------- */
 export type DataDeep = JSONData | CSVData | BIGWIGData | MultivecData | BEDDBData | VectorData | MatrixData | BAMData;
 
+/** values in the form of JSON */
 export interface Datum {
     [k: string]: number | string;
 }
 
+/**
+ * The JSON data format allows users to include data directly in the Gosling's JSON specification.
+ */
+
 export interface JSONData {
+    /**
+     * define data type
+     */
     type: 'json';
+
+    /** values in the form of JSON */
     values: Datum[];
+
+    /** specify the name of quantitative data fields */
     quantitativeFields?: string[];
+
+    /** specify the name of chromosome data fields */
     chromosomeField?: string;
+
+    /** specify the name of genomic data fields */
     genomicFields?: string[];
-    sampleLength?: number; // This limit the total number of rows fetched (default: 1000)
 
-    // !!! experimental
+    /** specify the number of rows loaded from the url.
+     *
+     * __Default:__1000
+     */
+    sampleLength?: number;
+
+    /** experimental */
     genomicFieldsToConvert?: {
         chromosomeField: string;
         genomicFields: string[];
     }[];
 }
 
+/**
+ * Any small enough tabular data files, such as tsv, csv, BED, BEDPE, and GFF, can be loaded using "csv" data specification.
+ */
+
 export interface CSVData {
     type: 'csv';
+
+    /**
+     * specify the URL address of the data file
+     */
     url: string;
+
+    /**
+     * specify file separator, __Default:__ ','
+     */
     separator?: string;
+
+    /**
+     * specify the name of quantitative data fields
+     */
     quantitativeFields?: string[];
+
+    /**
+     * specify the name of chromosome data fields
+     */
     chromosomeField?: string;
+
+    /**
+     * specify the name of genomic data fields
+     */
     genomicFields?: string[];
+
+    /**
+     * specify the number of rows loaded from the url.
+     *
+     * _Default:__ 1000
```

Comment:
```suggestion
     * __Default:__ `1000`
```

Missing `_` character broke the style of `Default`